docs/general/accounts.md

```diff
@@ -0,0 +1,112 @@
+# Accounts
+
+While the EVM defines two types of accounts (EOAs and CAs), Starkent only has
+one type of account. There is no distinction between accounts managed by a
+private key and a wallet (EOAs) and smart contracts, due to native Account
+Abstraction.
+
+Kakarot leverages this abstraction to provide a single account type that can be
+used for both EOA and CA use cases. This account type is managed by a single
+Starknet contract class (`account_contract.cairo`). This contract class defines
+the `__validate__` and `__execute__` entrypoints, which are used to send
+transactions from a wallet to Kakarot in a way that is compatible with the
+Starknet infrastructure. If you own the keys to an Ethereum address, you can use
+the `deploy_externally_owned_account` entrypoint to deploy a corresponding
+Starknet account contract, whose address is deterministic based on the Ethereum
+address.
+
+Although our backend makes no distinction between EOA and CA, the validation
+logic will reject any transaction that originates from a contract that has EVM
+code, as per the EVM specification. In the (extremely unlikely) event that a
+user sends a transaction from a contract account after mining the private key
+corresponding to an EOA, the transaction will be rejected.
+
+## Account storage
+
+Account contracts store the following information:
+
+- `storage`: A mapping from 32-byte keys to 32-byte values. This is used to
+  store the values stored in the EVM account storage.
+- `nonce`: A 64-bits value representing the number of transactions sent from the
```

Comment:
why only 64 bits?

Reply:
Starknet protocol limitation. In reality the nonce is also 64 bits on EVM clients, and attempting to send a tx with nonce >= 2**64 fails the EVM protocol rules.